src/filesystem.cc

```diff
@@ -891,10 +935,40 @@ GCSFileSystem::DeleteDirectory(const std::string& path)
 namespace as = azure::storage_lite;
 const std::string AS_URL_PATTERN = "as://([^/]+)/([^/?]+)(?:/([^?]*))?(\\?.*)?";
 
+struct ASCredential {
+  std::string account_str_;
+  std::string account_key_;
+
+  ASCredential();  // from env var
+  ASCredential(triton::common::TritonJson::Value& cred_json);
+};
+
+ASCredential::ASCredential()
+{
+  const auto to_str = [](const char* s) -> std::string {
+    return (s != nullptr ? std::string(s) : "");
+  };
+  const char* account_str = std::getenv("AZURE_STORAGE_ACCOUNT");
+  const char* account_key = std::getenv("AZURE_STORAGE_KEY");
+  account_str_ = to_str(account_str);
+  account_key_ = to_str(account_key);
+}
+
+ASCredential::ASCredential(triton::common::TritonJson::Value& cred_json)
+{
+  triton::common::TritonJson::Value account_str_json, account_key_json;
+  if (cred_json.Find("account_str", &account_str_json))
+    account_str_json.AsString(&account_str_);
+  if (cred_json.Find("account_key", &account_key_json))
+    account_key_json.AsString(&account_key_);
```

Comment:
If the JSON contains syntax error, the JSON parser will catch it before reaching this construction steps, and prints for example `error: creating server: Internal - failed to parse the request JSON buffer: Missing a comma or '}' after an object member. at 279`.

If an expected JSON object is of a different type, the JSON parser will not crash and skip forward. Then, credentials cannot be matched and `error: creating server: Not found - Cannot match credential for path  s3://s3-bucket-41498862/models1` is printed for example.

If the JSON is missing some parts of a credential, this is tricky because the Azure construction allows either/both `account_str` and `account_key` to be empty, and will use "default" if it is empty. If the user misses a key or sets it to empty, it might still be valid. The same goes for the S3 construction.

If the credential is invalid, the `check_client()` will catch it, for example with [`log-verbose=1` and S3 `"region": 0`] prints the following:

```
I0802 06:04:24.262804 33401 filesystem.cc:2335] Using credential    for path  s3://s3-bucket-41498862/models1
I0802 06:04:28.612361 33401 metrics.cc:650] Collecting metrics for GPU 0: NVIDIA TITAN V
I0802 06:04:28.612790 33401 tritonserver.cc:2176] ...(skip the table printout)

I0802 06:04:28.612829 33401 server.cc:259] No server context available. Exiting immediately.
error: creating server: Internal - Unable to create S3 filesystem client. Check account credentials.
+ exit 1
```
Basically, accessing the S3 path `s3://s3-bucket-41498862/models1` is using the default/empty S3 credential and gets account credential error, which should signals for checking for the default S3 credential. The only possible issue is the user must remember to set `log-verbose` to print which credential is used for which path.